internal/persistence/sql/relationtuples.go

```diff
@@ -260,15 +272,63 @@ func (p *Persister) ExistsRelationTuples(ctx context.Context, query *relationtup
 	return exists, sqlcon.HandleError(err)
 }
 
+func buildInsert(commitTime time.Time, nid uuid.UUID, rs []*relationtuple.RelationTuple) (query string, args []any, err error) {
+	if len(rs) == 0 {
+		return "", nil, errors.WithStack(ketoapi.ErrMalformedInput)
+	}
+
+	var q strings.Builder
+	fmt.Fprintf(&q, "INSERT INTO %s (shard_id, nid, namespace, object, relation, subject_id, subject_set_namespace, subject_set_object, subject_set_relation, commit_time) VALUES ", (&RelationTuple{}).TableName())
+	const placeholders = "(?, ?, ?, ?, ?, ?, ?, ?, ?, ?)"
+	const separator = ", "
+	q.Grow(len(rs) * (len(placeholders) + len(separator)))
+	args = make([]any, 0, 10*len(rs))
+
+	for i, r := range rs {
+		if r.Subject == nil {
+			return "", nil, errors.WithStack(ketoapi.ErrNilSubject)
+		}
+
+		rt := &RelationTuple{
+			ID:         uuid.Must(uuid.NewV4()),
```

Comment:
Use `gen_random_uuid()` on crdb?

Reply:
We could, but performance is not likely to increase a lot because these are global tables anyway.